packages/test/test-end-to-end-tests/src/test/directoryEndToEndTests.spec.ts

```diff
@@ -751,6 +753,9 @@ describeNoCompat("SharedDirectory orderSequentially", (getTestObjectProvider) =>
         sharedDir.on("subDirectoryDeleted", (path, _local, _target) => {
             subDirDeletedEventData.push(path);
         });
+        sharedDir.on("subDirectoryUndeleted", (path, _local, _target) => {
+            subDirUndeletedEventData.push(path);
+        });
```

Comment:
For now, the tests will fail because I haven't actually used these in the test cases. Once I determine if the implementation I have gone with is correct, I will update the test cases to assert against the new undeleted event.